src/core/hle/kernel/k_thread.h

```diff
@@ -752,4 +771,16 @@ class KThread final : public KAutoObjectWithSlabHeapAndContainer<KThread, KSynch
     }
 };
 
+class KScopedDisableDispatch {
+public:
+    explicit KScopedDisableDispatch(KernelCore& kernel_) : kernel{kernel_} {
```

Comment:
```suggestion
    [[nodiscard]] explicit KScopedDisableDispatch(KernelCore& kernel_) : kernel{kernel_} {
```

Will allow the compiler to flag logic bugs where an instance is constructed and destroyed immediately.

e.g. accidentally doing

```cpp
{
    KScopedDisableDispatch{kernel}
    // more code...
}
```

instead of:

```cpp

{
    KScopedDisableDispatch dd{kernel}
    // more code...
}
```